src/Compilers/CSharp/Test/Syntax/IncrementalParsing/IncrementalParsingTests.cs

```diff
@@ -2341,6 +2341,75 @@ public void InsertOpenBraceBeforeCodes()
             CompareIncToFullParseErrors(reparsedTree, parsedTree);
         }
 
+        [WorkItem(6676, "https://github.com/dotnet/roslyn/issues/6676")]
+        [Fact]
+        public void InsertExpressionStatementWithoutSemicolonBefore()
+        {
+            SourceText oldText = SourceText.From(@"
+System.Console.WriteLine(true)
+");
+            var startTree = SyntaxFactory.ParseSyntaxTree(oldText, options: TestOptions.Script);
+
+            startTree.GetDiagnostics().Verify();
+
+            var newText = oldText.WithChanges(new TextChange(new TextSpan(0, 0), "\nSystem.Console.WriteLine(false)"));
+            var reparsedTree = startTree.WithChangedText(newText);
+            var parsedTree = SyntaxFactory.ParseSyntaxTree(newText, options: TestOptions.Script);
+
+            parsedTree.GetDiagnostics().Verify(
+                // (2,33): error CS1002: ; expected
+                // System.Console.WriteLine(true)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(2, 32));
```

Comment:
>Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(2, 32) [](start = 16, length = 67)

It is not clear why we are getting an error on the last line, I would expect the error to be on the previous line, the one that the test is adding.